src/Core/Form/IdentifiableObject/DataHandler/CountryFormDataHandler.php

```diff
@@ -0,0 +1,141 @@
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(CommandBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CountryConstraintException
+     */
+    public function create(array $data)
+    {
+        $addCountryCommand = new AddCountryCommand(
+            $data['country'],
+            $data['iso_code'],
+            (int) $data['call_prefix'],
+            $data['address_format']
+        );
+
+        if (null !== $data['zip_code_format']) {
+            $addCountryCommand->setZipCodeFormat($data['zip_code_format']);
+        }
+
+        if (null !== $data['default_currency']) {
+            $addCountryCommand->setDefaultCurrency($data['default_currency']);
+        }
+
+        if (null !== $data['zone'] && !empty($data['zone'])) {
+            $addCountryCommand->setZone($data['zone']);
+        }
+
+        if (null !== $data['need_zip_code']) {
+            $addCountryCommand->setNeedZipCode($data['need_zip_code']);
+        }
+
+        if (null !== $data['is_enabled']) {
+            $addCountryCommand->setEnabled($data['is_enabled']);
+        }
+
+        if (null !== $data['contains_states']) {
+            $addCountryCommand->setContainsStates($data['contains_states']);
+        }
+
+        if (null !== $data['need_identification_number']) {
+            $addCountryCommand->setNeedIdNumber($data['need_identification_number']);
+        }
+
+        if (null !== $data['display_tax_label']) {
+            $addCountryCommand->setDisplayTaxLabel($data['display_tax_label']);
+        }
+
+        if (null !== $data['shop_association'] && !empty($data['shop_association'])) {
+            $addCountryCommand->setShopAssociation($data['shop_association']);
+        }
+
+        /** @var CountryId $countryId */
+        $countryId = $this->commandBus->handle($addCountryCommand);
+
+        return $countryId->getValue();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CountryConstraintException
+     */
+    public function update($countryId, array $data)
+    {
+        $command = new EditCountryCommand(
+            (int) $countryId,
+            $data['country'],
+            $data['iso_code'],
+            (int) $data['call_prefix'],
+            $data['zone'],
+            $data['need_zip_code'],
+            $data['address_format'],
+            $data['is_enabled'],
+            $data['contains_states'],
+            $data['need_identification_number'],
+            $data['display_tax_label'],
+            $data['default_currency']
+        );
+
+        if (null !== $data['zip_code_format']) {
+            $command->setZipCodeFormat($data['zip_code_format']);
+        }
+
+        if (null !== $data['shop_association']) {
+            $command->setShopAssociation($data['shop_association']);
+        }
+
+        $this->commandBus->handle($command);
```

Comment:
No return ?

Reply:
If you check all the others handlers firing **Edit**NameCommand - they all do not return anything